### ISSUE TITLE
Last stacked window gets unstacked automatically

### ISSUE THREAD

**Issue by jvansant** [pop-os/shell#1301]
When I close out of the second to last window in a stack the last remaining window is left unstacked. This is very aggravating as I have to create a new stack every time I open a new window. Otherwise the new window is not stacjed and is instead tiled as it would be normally. 

Here is a previous issue citing the same occurrence:

I can confirm this; the last window remains in stack mode if you remove the penultimate window from the stack, however closing that window destroys the stack. 

Unfortunately the commit doesn't fix the issue. 

-- bvs

_Originally posted by @bettyvschmartz in https://github.com/pop-os/shell/issues/881#issuecomment-886538473_

**Comment by mmstick** [comment]
It's currently expected behavior to unstack the last window. Unsure on what the general consensus is for the best default behavior.

**Comment by jacobgkau** [comment]
It might at least make sense to have keyboard closing a window not unstack the remaining window, since keyboard moving a window doesn't unstack the remaining window. The only reason the auto-unstack was added (as far as I know) was because there's no way to explicitly unstack a single window using the mouse.

**Comment by mmstick** [comment]
Yeah, I need to investigate this again to see if I can intercept that easier now.

**Comment by bettyvschmartz** [comment]
> It's currently expected behavior to unstack the last window. Unsure on what the general consensus is for the best default behavior.

I still struggle to understand why this is expected behavior and why the change was made in the first place. In the sense that I understand;

> the auto-unstack was added (as far as I know) was because there's no way to explicitly unstack a single window using the mouse.

But this seems like a hack to account for a technical limitation around mouse usage but makes no sense IMO from a UX perspective. Given that stacking is a toggle (on/off) by the user you would expect disabling should also be actioned by the user. Automatically toggling off makes it appear broken, i.e. it's unexpected behaviour. 

For example;

- One enables stacking (super+s)
- Opens a few windows of desired stacked app as tabs
- Closes each tab in the stack when finished with them (except for the parent/main window)
- Proceeds to open some new windows expecting tabs
- Gets separate windows

This suddenly differing behaviour is confusing especially if the primary usage is managing child windows of a parent app. Maybe it makes a little more sense if one's stacks are across apps so as you close all apps in a stack, the remaining app is unstacked but I doubt this is primary usage for stacking. 

It's always best to design for sane defaults. This change in behaviour isn't that, and leaving stacking enabled doesn't cause confusion in either use case I outlined (across apps, or for child windows). What's the worst case for those who use stacking across apps? They're left with an orange title bar on their last remaining window? Seems the least confusing/awkward/problematic of the two approaches. 

Currently, I have to leave a random extra window open to retain stacking or else it's required to continually hit super+s and move new windows back into the stack. 

-- bvs

**Comment by pavangayakwad** [comment]
It would be great to have an option in the settings, "always maintain stacked windows" to avoid surprising unstack of windows without users consent.

**Comment by pavangayakwad** [comment]
> It's currently expected behavior to unstack the last window. Unsure on what the general consensus is for the best default behavior.

When I can stack a single window to start the stacking, It is expected to maintain the same state when sibling windows were closed. Unstacking suddenly without users consent is a surprising behavior that user hardly appreciate.

**Comment by bettyvschmartz** [comment]
It's looking less and less likely to be fixed to be honest because...

> It's currently expected behavior to unstack the last window

But as you say;

> Unstacking suddenly without users consent is a surprising behavior that user hardly appreciate.

It's just a really odd UX decision to take. 

-- bvs
